src/main/java/com/client/config/JpaConfig.java

```diff
@@ -0,0 +1,48 @@
+package com.client.config;
+
+import com.client.ApplicationSettings;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.boot.jdbc.DataSourceBuilder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
+import org.springframework.orm.jpa.JpaTransactionManager;
+import org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean;
+import org.springframework.orm.jpa.support.SharedEntityManagerBean;
+import org.springframework.orm.jpa.vendor.HibernateJpaVendorAdapter;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+
+import javax.sql.DataSource;
+
+@Configuration
+public class JpaConfig {
+    @Bean
+    public LocalContainerEntityManagerFactoryBean entityManagerFactory(ApplicationSettings appSettings, DataSource dataSource) {
```

Comment:
@scriptom 
What is the purpose of using an interface here as the return type here? There's a long list:
FactoryBean<EntityManagerFactory>, BeanClassLoaderAware, BeanFactoryAware, BeanNameAware, InitializingBean, DisposableBean, EntityManagerFactoryInfo, PersistenceExceptionTranslator, Serializable

There are also a number of interfaces extending other interfaces here. Did you have one in mind that you thought could encapsulate future behavior, because none really pop out to me.

I guess we could extend the abstract class?
AbstractEntityManagerFactoryBean

Reply:
@taswartz I thought it had a friendlier interface we could've put in its place. Let's leave it as it is now.